src/components/webview/FileExplorerProvider.ts

```diff
@@ -74,47 +75,67 @@ export class FileExplorerProvider implements WebviewViewProvider {
 		this.__webviewResolver = new WebviewResolver(this.__extensionPath);
 	}
 
-	refresh(): void {
-		if (!this.__view) {
-			return;
-		}
-
-		this.__webviewResolver.resolveWebview(
-			this.__view.webview,
-			'fileExplorer',
-			'{}',
-		);
-	}
-
 	resolveWebviewView(webviewView: WebviewView): void | Thenable<void> {
-		if (!webviewView.webview) {
-			return;
-		}
+		this.__view = webviewView;
+
+		const viewProps = this.__buildViewProps(this.__lastSelectedCaseHash);
 
 		this.__webviewResolver.resolveWebview(
 			webviewView.webview,
 			'fileExplorer',
-			'{}',
+			JSON.stringify({
+				viewProps,
+			}),
 		);
-		this.__view = webviewView;
 
 		this.__view.onDidChangeVisibility(() => {
-			if (this.__lastSelectedCaseHash === null) {
-				return;
-			}
 			// display folders/files for the lastly selected case when panel is collapsed and re-opened
-			this.updateExplorerView(this.__lastSelectedCaseHash);
+			this.setView(this.__lastSelectedCaseHash);
+
+			this.__webviewResolver.resolveWebview(
+				webviewView.webview,
+				'fileExplorer',
+				JSON.stringify({
+					viewProps,
+				}),
+			);
 		});
 
 		this.__attachExtensionEventListeners();
 		this.__attachWebviewEventListeners();
 	}
 
-	public setView(data: Extract<View, { viewId: 'fileExplorer' }>) {
-		this.__lastData = data;
+	public setView(caseHash: CaseHash | null): void {
```

Comment:
In the future, `setView` should accept no parameters (because the only goal is to send information to the view). For not it's fine.